packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,216 @@
 import { Controller } from 'ember-runtime';
-import { run } from 'ember-metal';
-
-import { Application } from 'ember-application';
-import { Router } from 'ember-routing';
-import { compile } from 'ember-template-compiler';
-import {
-  Component,
-  setTemplates,
-  setTemplate
-} from 'ember-glimmer';
-import { jQuery } from 'ember-views';
-
-let App, appInstance;
+import { Component } from 'ember-glimmer';
+import { moduleFor, AutobootApplicationTestCase } from 'internal-test-helpers';
 
-function prepare() {
-  setTemplate('components/expand-it', compile('<p>hello {{yield}}</p>'));
-  setTemplate('application', compile('Hello world {{#expand-it}}world{{/expand-it}}'));
-}
+moduleFor('Application Lifecycle - Component Registration', class extends AutobootApplicationTestCase {
 
-function cleanup() {
-  run(() => {
-    try {
-      if (App) {
-        App.destroy();
-      }
-      App = appInstance = null;
-    } finally {
-      setTemplates({});
-    }
-  });
-}
-
-QUnit.module('Application Lifecycle - Component Registration', {
-  setup: prepare,
-  teardown: cleanup
-});
+  ['@test The helper becomes the body of the component'](assert) {
+    this.runTask(() => {
+      this.createApplication();
 
-function boot(callback, startURL = '/') {
-  run(() => {
-    App = Application.create({
-      name: 'App',
-      rootElement: '#qunit-fixture'
+      this.addTemplate('components/expand-it', '<p>hello {{yield}}</p>');
+      this.addTemplate('application', 'Hello world {{#expand-it}}world{{/expand-it}}');
     });
 
-    App.deferReadiness();
-
-    App.Router = Router.extend({
-      location: 'none'
-    });
-
-    appInstance = App.__deprecatedInstance__;
-
-    if (callback) { callback(); }
-  });
-
-  let router = appInstance.lookup('router:main');
+    this.visit('/').then(() => {
```

Comment:
In fact I think many of the visit calls here are not needed. Because this is an "autoboot" test, the app will be booted and rendered immediate after the `runTask` wrapping `createApplication`